pennylane/utils.py

```diff
@@ -353,3 +356,53 @@ def __str__(self):
                 output += "{}({}(wires={}))\n".format(return_map[op.return_type], op.name, op.wires)
 
         return output
+
+
+def inv(operation_list):
+    """Invert a list of operations or a template.
+
+    If the inversion happens inside a QNode, the operations are removed and requeued
+    in the reversed order for proper inversion.
```

Comment:
This queuing system really does make it difficult sometimes :laughing:

Reply:
And it definitely needs a rework!